### ISSUE TITLE
A false negative about the rule Indentation

### ISSUE THREAD

**Issue by clearlove2077** [checkstyle/checkstyle#13782]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ cat T.java
import util.MyAnnotation;
@MyAnnotation
public @Ann
@Ann2 class T
    implements TestInterface {
}

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="charset" value="UTF-8"/>
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="2"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>



/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-10.12.3-all.jar -c config.xml T.java
Starting audit...
Audit done.
```

I would expect Checkstyle to show the warning message like:
```bash
Starting audit...
[WARN] T.java:4:1: '@' has incorrect indentation level 0, expected level should be 4. [Indentation]
Audit done.
```
but no warnings, hence, I think this is an FN.

**Comment by romani** [comment]
This is "Line wrap" should be +4 indentation.

**Comment by clearlove2077** [comment]
> This is "Line wrap" should be +4 indentation.

Hi @romani , I want to provide another case for helping analysis, CheckStyle can report an Indention warning at line 2, but I use **the same config file**.

```java
public @Ann
@Ann2 class T  // violation
    implements TestInterface {
}
```

Warning message:
```bash
[WARN] T.java:2:1: '@' has incorrect indentation level 0, expected level should be 4. [Indentation]
```

**Comment by kalpadiptyaroy** [comment]
@clearlove2077 - Kind request - Can you kindly edit and modify your issue description as follows. Since I see your sample code has a bit of weird java syntax!

**Old code**
```
public @Ann
@Ann2 class T  // violation
    implements TestInterface {
}
```

**New Code**

```
@Ann
@Ann2 public class T  // violation
    implements TestInterface {
}
```

Since as far as I know annotations can't occur between `public` keyword and `class` keyword!
I am well aware that this doesn't affect the merits of the current issue. But it helps to avoid confusion for contributors!

**Comment by nrmancuso** [comment]
> Since as far as I know annotations can't occur between public keyword and class keyword!

@kalpadiptyaroy no need to speculate, please prove your claim by javac.

**Comment by kalpadiptyaroy** [comment]
@nrmancuso - My myth is busted. It is legal to use public before annotations while annotating a class. Removing my previous comment.

**Comment by kalpadiptyaroy** [comment]
I am on it. I have written a basic test case in the following PR. Please have a look
